### ISSUE TITLE
Error message for not allowed queryoptions should not contain information on how to enable it.

### ISSUE THREAD

**Issue by dnperfors** [OData/AspNetCoreOData#731]
**Assemblies affected**
ASP.NET Core OData 8.0.11

**Describe the bug**
When disabling QueryOptions, for example by setting AllowedQueryOptions = AllowedQueryOptions.None, the error message now adds the following to the error message: "To allow it, set the 'AllowedQueryOptions' property on EnableQueryAttribute or QueryValidationSettings."
Since this error is (by default) send to the client, it doesn't make sense to add this message and thus should be removed. The client is not able to enable these features.

**Comment by julealgon** [comment]
Can you elaborate where this message is currently coming from? Does the server throw a raw exception with the message, or is the message output as text as part of some specific http error other than the standard exception-based 500?

The reason I'm asking is because, if that message is part of an exception that is output directly, then your argument doesn't hold IMHO: preventing exception details from reaching customers is an application concern, not a library concern.

Now, if OData is catching the exception and converting it to an error message automatically and reusing the exception message, then I agree this needs changing (not the message, but the conversion itself).

Lastly, this same type of message is used in a few different validation errors and are not limited to query options:
- https://github.com/OData/AspNetCoreOData/blob/f675f6c6abe2cd10bfa00ce8c4f7dc83b49e1293/src/Microsoft.AspNetCore.OData/Properties/SRResources.resx#L399-L401
- https://github.com/OData/AspNetCoreOData/blob/f675f6c6abe2cd10bfa00ce8c4f7dc83b49e1293/src/Microsoft.AspNetCore.OData/Properties/SRResources.resx#L417-L425
- https://github.com/OData/AspNetCoreOData/blob/f675f6c6abe2cd10bfa00ce8c4f7dc83b49e1293/src/Microsoft.AspNetCore.OData/Properties/SRResources.resx#L453-L455

**Comment by dnperfors** [comment]
I noticed the message on the response itself, so it looks like the library is automatically converting the ODataException that is thrown by the validation directly. Besides that, I believe I saw that the context of what operation the exception is talking about is lost, so you only have the exception message. I could be wrong though, since I currently can't check myself.

**Comment by dnperfors** [comment]
A small update, I modified the PeopleController as described earlier and did the following request:
`http://localhost:5000/People?$compute=substring(Firstname,%201,%205)%20as%20test`
I then got the following response:
```
{
  "error": {
    "code": "",
    "message": "The query specified in the URI is not valid. Query option 'Compute' is not allowed. To allow it, set the 'AllowedQueryOptions' property on EnableQueryAttribute or QueryValidationSettings.",
    "details": [],
    "innererror": {
      "message": "Query option 'Compute' is not allowed. To allow it, set the 'AllowedQueryOptions' property on EnableQueryAttribute or QueryValidationSettings.",
      "type": "Microsoft.OData.ODataException",
      "stacktrace": "   at Microsoft.AspNetCore.OData.Query.Validator.ODataQueryValidator.ValidateQueryOptionAllowed(AllowedQueryOptions queryOption, AllowedQueryOptions allowed) in C:\\dev\\AspNetCoreOData\\src\\Microsoft.AspNetCore.OData\\Query\\Validator\\ODataQueryValidator.cs:line 141\r\n   at Microsoft.AspNetCore.OData.Query.Validator.ODataQueryValidator.Validate(ODataQueryOptions options, ODataValidationSettings validationSettings) in C:\\dev\\AspNetCoreOData\\src\\Microsoft.AspNetCore.OData\\Query\\Validator\\ODataQueryValidator.cs:line 41\r\n   at Microsoft.AspNetCore.OData.Query.ODataQueryOptions.Validate(ODataValidationSettings validationSettings) in C:\\dev\\AspNetCoreOData\\src\\Microsoft.AspNetCore.OData\\Query\\ODataQueryOptions.cs:line 634\r\n   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.ValidateQuery(HttpRequest request, ODataQueryOptions queryOptions) in C:\\dev\\AspNetCoreOData\\src\\Microsoft.AspNetCore.OData\\Query\\EnableQueryAttribute.cs:line 707\r\n   at Microsoft.AspNetCore.OData.Query.EnableQueryAttribute.OnActionExecuting(ActionExecutingContext actionExecutingContext) in C:\\dev\\AspNetCoreOData\\src\\Microsoft.AspNetCore.OData\\Query\\EnableQueryAttribute.cs:line 167"
    }
  }
}
```

There is an ODataException thrown at: https://github.com/OData/AspNetCoreOData/blob/main/src/Microsoft.AspNetCore.OData/Query/Validator/ODataQueryValidator.cs#L141

Which is translated into a response at: https://github.com/OData/AspNetCoreOData/blob/main/src/Microsoft.AspNetCore.OData/Query/EnableQueryAttribute.cs#L190

**Comment by julealgon** [comment]
Yeah, if there is no option to omit the message on the translation step, then I agree this is bad.

Same goes for the stacktrace, BTW.

The same mechanisms that exist to hide exception details from customers should be considered for these automatic response messages as well.

**Comment by habbes** [comment]
@dnperfors thanks for reporting this. @julealgon makes a good point about providing mechanisms for the customers to handle these errors or customize/override the automatic response.
